openslides/utils/views.py

```diff
@@ -45,3 +47,20 @@ def method_call(self, request: Any, *args: Any, **kwargs: Any) -> Any:
     # Add the http-methods and delete the method "method_call"
     get = post = put = patch = delete = head = options = trace = method_call
     del method_call
+
+
+class IndexView(View):
+    """
+    A view to serve a single cached template file. Subclasses has to provide 'template_name'.
+    """
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+        if 'template_name' not in dir(self):
+            raise AttributeError("'template_name' is not provided")
```

Comment:
I think the [ImproperlyConfigured](https://docs.djangoproject.com/en/1.11/ref/exceptions/#improperlyconfigured) Exception is better here.

I other places we don't use dir(self) but set the needed attribute as class attribute set to None. I think this helps mypy. So for example:
```python
class IndexView(View):
    template_name = None  # type: str
    def foo(...):
        if self.template_name is None:
            raise ImproperlyConfigured(...)
```